src/web/TextInput.tsx

```diff
@@ -122,12 +122,29 @@ export class TextInput extends React.Component<Types.TextInputProps, TextInputSt
                     onMouseUp={ this._checkSelectionChanged }
                     onPaste={ this._onPaste }
                     aria-label={ this.props.accessibilityLabel }
-                    type={ this.props.secureTextEntry ? 'password' : 'text' }
+                    type={ this.props.secureTextEntry ? 'password' : this._keyboardType() }
                 />
             );
         }
     }
 
+    private _keyboardType = () => {
+        // Show the correct virtual keyboardType in HTML 5
+        var keyboardType = this.props.keyboardType || 'text';
```

Comment:
I recommend not initializing keyboardType with this.props.keyboardType. It's safer to always initialize it to the default value 'text'. Then override the default if this.props.keyboardType is one of the known overrides. With the current implementation, if a new keyboardType is added to RN, it will cause this code to break.